src/libraries/Native/Unix/System.Native/pal_console.c

```diff
@@ -182,8 +182,17 @@ static bool ConfigureTerminal(bool signalForBreak, bool forChild, uint8_t minCha
     {
         termios.c_iflag &= (uint32_t)(~(IXON | IXOFF));
         termios.c_lflag &= (uint32_t)(~(ECHO | ICANON | IEXTEN));
+
+        if (disableCrLfConversions)
+        {
+            // Disable any CR-to-NL, NL-to-CR or Ignore-CR settings
+            // Required for scenaria where the precise input character is needed
+            // such as System.Console.ReadKey()
+            termios.c_iflag &= (uint32_t)(~(ICRNL | INLCR | IGNCR));
```

Comment:
suggestion:
```
termios.c_iflag &= (uint32_t)(~(IXON | IXOFF | ICRNL | INLCR | IGNCR));
if (!disableCrLfConversions)
{
     termios.c_iflag |= ICRNL;
}
termios.c_lflag &= ...
```
and maybe rename `!disableCrLfConversions` -> `convertCrToNl`.

Reply:
Addressed.